src/sync.cpp

```diff
@@ -95,6 +95,36 @@ Sync::Sync(MegaClient* cclient, string* crootpath, const char* cdebris,
     localroot.init(this, FOLDERNODE, NULL, crootpath);
     localroot.setnode(remotenode);
 
+#ifdef __APPLE__
+    if (macOSmajorVersion() >= 19) //macOS catalina+
+    {
+        LOG_debug << "macOS 10.15+ filesystem detected. Checking fseventspath.";
+        string supercrootpath = "/System/Volumes/Data" + *crootpath;
+
+        int fd = open(supercrootpath.c_str(), O_RDONLY);
+        if (fd == -1)
+        {
+            LOG_debug << "Unable to open path using fseventspath.";
+            this->fsEventsPath = *crootpath;
```

Comment:
New class members should be named with a `m` prefix. Please, refactor `fsEventsPath` to `mFsEventsPath` or alike (and you should skip also the usage of `this->` to access the member (we avoid clashes with other variables thanks to the prefix).

Reply:
Will do. Thanks.